src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs

```diff
@@ -0,0 +1,232 @@
+            yield return new[] { TexlStrings.SummarizeArg1, TexlStrings.SummarizeArg2, TexlStrings.SummarizeArg2, TexlStrings.SummarizeArg3, TexlStrings.SummarizeArg3 };
+        }
+
+        // Produce a signature of the form:
+        // Sumamrize(source, grouping_column, aggregation, grouping_column, aggregation, ...)
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            Contracts.Assert(arity >= 0);
+
+            if (arity <= 3)
+            {
+                return base.GetSignatures(arity);
+            }
+
+            // Limit the argCount avoiding potential OOM
+            int argCount = arity > SignatureConstraint.RepeatTopLength + SignatureConstraint.EndNonRepeatCount ? SignatureConstraint.RepeatTopLength + SignatureConstraint.EndNonRepeatCount : arity;
+            var args = new TexlStrings.StringGetter[argCount];
+
+            int lastIndex = argCount - 1;
+            args[0] = TexlStrings.SummarizeArg1;
+
+            var addSwitch = false;
+
+            for (int i = 1; i < lastIndex; i++)
+            {
+                if (addSwitch)
+                {
+                    args[i] = TexlStrings.SummarizeArg2;
+                }
+                else
+                {
+                    args[i] = TexlStrings.SummarizeArg3;
+                }
+
+                addSwitch = !addSwitch;
+            }
+
+            return new[] { args };
+        }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            const string thisGroup = "ThisGroup";
+
+            bool isValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+            Contracts.Assert(returnType.IsTable);
+
+            // Source table type
+            DType sourceType = argTypes[0];
+
+            if (args[0] is AsNode)
+            {
+                isValid = false;
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSummarizeDataSourceScopeNotSupported);
+            }
+
+            if (sourceType.Contains(new DName(thisGroup)))
+            {
+                isValid = false;
+                errors.EnsureError(DocumentErrorSeverity.Severe, args[0], TexlStrings.ErrSummarizeDataSourceContainsThisGroupColumn);
+            }
+
+            var atLeastOneGroupByColumn = false;
+
+            for (int i = 1; i < args.Length; i++)
```

Comment:
>for [](http://example.com/codeflow?start=12&length=3)

If I understand this loop correctly, it will allow something like `Summarize(MyCollection, CountRows(ThisGroup) As count, Brand, Sum(ThisGroup, Price) As TotalPrice, Year)`, which is more than we said we would do on the discussion. We should block that.

Reply:
Are you concerned about the args order? If that's the case lets catch up offline.

Reply:
what was resolution?